core/spec/models/spree/order_shipping_spec.rb

```diff
@@ -138,7 +138,7 @@ def emails
       before { Spree::OrderCancellations.new(order).short_ship([order.inventory_units.first]) }
 
       it "updates the order shipment state" do
-        expect { subject }.to change { order.reload.shipment_state }.from('ready').to('shipped')
+        expect { subject }.to change { order.reload.shipment_state }.from('pending').to('shipped')
```

Comment:
If that case it was previously broken. Previously the shipment went to the `ready` state only because it was checking if inventory units where "not backordered". But just because the shipment was once again ready the individual inventory units where still in the "canceled" state and there is not path from "canceled" to shipped because `allow_ship?` is only true if on-hold.

This is actually a good example of why having the logic in two places (and be different) is problematic. On `master` if you cancel an item then try to ship it you will likely get an error because you cannot `ship` the inventory unit anymore since it is canceled and not on-hold.

With my new code the logic has a single source of truth which means since you cannot ship the inventory unit the shipment is also moved back to pending. If you want to make the spec pass as originally designed then we need to update the `allow_ship?` method in InventoryUnit to be `on_hand? || canceled?`.

If that sounds good I'll make the change and revert this test to it's original check.